### PR TITLE
feat: adding mercantile bank scraper

### PR DESCRIPTION
**Pull request by EzzatQ**
Changes:

* Created a new scraper for bank Mercantile based on the Discount bank. The implementations are identical as both mercantile and discount use the same system. The only difference between the two is the login url (`https://start.telebank.co.il/login/?bank=m`).  Code is based on [pull request](https://github.com/eshaham/israeli-bank-scrapers/pull/599) by @kfirarad.

* I have a mercantile account and I was able to test the scrapper successfully. Manually inspecting the payload returned by `scraper.scrape`:

```
{
  "success": true,
  "accounts": [
    {
      "accountNumber": "**redacted**",
      "balance":"**redacted**,
      "txns": [
        {
          "type": "normal",
          "identifier": "**redacted**",
          "date": "2024-02-01T22:00:00.000Z",
          "processedDate": "2024-01-30T22:00:00.000Z",
          "originalAmount": -20,
          "originalCurrency": "ILS",
          "chargedAmount": -20,
          "description": ""**redacted**",
          "status": "completed"
        },
        ...
      ]
    }
  ]
}
```

### CODE REVIEW

**Comment by baruchiro** [comment]
Welcome @EzzatQ !!

I hope I will review it soon :-)

**Comment by EzzatQ** [comment]
I continued testing this and it looks like we can only extract up to one year of transactions from current day (or actually 364 days to be exact), if requesting a longer period, mercantile's api will return an error message saying that "the requested information is not available, if you need more assistance call us on ....".

I'm wondering whether I should add some validation to the `options.startDate` param to avoid unexpected behavior. I can either throw an exception with a matching message, or take the maximum date between the passed `options.startDate` and `currentDate - 364`. What do you think?

**Comment by baruchiro** [comment]
@EzzatQ see the Discount bank as an example of limiting the `startDate`. It is not the only scraper that's doing the same technique:

https://github.com/eshaham/israeli-bank-scrapers/blob/74479a1d9537039098d5f714471d6926da897b45/src/scrapers/discount.ts#L78-L80

**Comment by baruchiro** [comment]
Also, I will add you to the Code Owners once we merge this, is that OK for you?
https://github.com/eshaham/israeli-bank-scrapers/issues/830

**Comment by EzzatQ** [comment]
I am aware of the defaultStartMoment solution, in fact it is implemented in the mercantile scraper because it extends the discount scraper class. What I am talking about is validation of the user input, because if I supply a date that is before 1 year prior, I get an error from mercantile api, and that error is not propagated to the user. Even if it was, the error tells me nothing about why that issue happened aka the user won't know that the scraper is failing because of an invalid start date.

I personally have no problem to be added as a Code Owner. I don't know about @kfirarad as his PR is 3 years old

**Comment by baruchiro** [comment]
Correct me if I'm wrong, but if you use the **max** date between the user `startDate` input and the _one year ago_ date, the start date will never go far beyond one year, so you will never get the error from the bank, right?

**Comment by kfirarad** [comment]
Hi!
@EzzatQ Thanks for taking care of this one. I never had the chance to move it forward.
@baruchiro Feel free to add me as a code owner. Thank you for the suggestion.

**Comment by baruchiro** [comment]
> Yes exactly

@EzzatQ So I don't understand when you will get the error, there will be no case for retrieving more than one year ago.

File: package-lock.json
Comment (baruchiro) [inline package-lock.json:4]: If you didn't changed the `package.json` the `package.lock.json` should not be changed too.

I guess it is because you're using the wrong version of NodeJS, which needs to be `v16.19.0`.

File: src/scrapers/mercantile.ts
Comment (baruchiro) [inline src/scrapers/mercantile.ts:13]: OMG I postponed this review for long time, just to finally review it and see it based on another scraper??

Sorry!!

**Review by baruchiro — APPROVED** [review]
Approved!

I see it is the same scraper like DiscountScraper. If it is working for you, I accept it.

**Comment by baruchiro** [comment]
Status: I think it is a breaking change, I waiting for @eshaham @esakal

**Comment by EzzatQ** [comment]
Sorry for disappearing, got very busy.
Thanks for reviewing and merging!

**Pull request merged**
